b2g/components/RemoteControlService.jsm

```diff
@@ -175,6 +208,15 @@ this.RemoteControlService = {
     // Monitor xpcom-shutdown to stop service and clean up
     Services.obs.addObserver(this, "xpcom-shutdown", false);
 
+    // Listen control mode change from gaia
+    SystemAppProxy.addEventListener("mozContentEvent", this);
```

Comment:
need to move event listener while service stop. In addition, you'll need to sync the `isCursorMode` from Gaia while service start since you can turn enable the service while browser app is active.